packages/components/src/FilterBar/context/reducer/updateSingleFilter.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { FiltersState } from "../types"
+import { baseFilterAttributes } from "./__testdata"
+import { filtersStateReducer } from "./filtersStateReducer"
+
+type Values = {
+  flavour: string
+  sugarLevel: number
+}
+
+const stateFilters = {
+  flavour: {
+    ...baseFilterAttributes,
+    id: "flavour",
+    name: "Flavour",
+  },
+  sugarLevel: {
+    ...baseFilterAttributes,
+    id: "sugarLevel",
+    name: "Sugar Level",
+    isRemovable: true,
+    isActive: false,
+  },
+} satisfies FiltersState<Values>["filters"]
+
+describe("filtersStateReducer: update_single_filter", () => {
+  it("updates state of a single filter", () => {
+    const state = {
+      filters: stateFilters,
+      activeFilterIds: new Set<keyof Values>(["flavour"]),
+    } satisfies FiltersState<Values>
+
+    expect(
```

Comment:
An observation only: I'm sure this test does properly test the functionality, but it's hard for me when reading it do understand really what exactly is going on here and compared.
It's using complex objects, splatting external state in, etc. - not very obvious.

If I'm reading this correctly, it's all about setting `isOpen` to `true` for the `Flavour` filter?

Could this be
- Assert `isOpen` is `false`
- Invoke reducer
- Assert `isOpen` is true
?
I get that it compares a lot more, and thus might be 'safer', but might not be the responsibility of this test either?

Reply:
Hmm... Yeah I was going for the safer option, thus testing it as a whole.

Unless I add a test that takes in the state with no `data` and ensure it comes back exactly the same? The main thing is I want to ensure nothing goes missing in between.

In the case of the Active State, that does a little more than changing just the `filters` (`activeFilterIds` get updated too).

If there's at least one test that ensures things don't disappear (eg. if I go to update Flavour, I don't accidentally delete SugarLevel by forgetting to spread the rest of the filters back in), then I could simplify the others.

Thoughts?